File: BOJ/10773_제로/go/Main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	var count int
	var stack []int
	total := 0
	fmt.Scan(&count)
	for i := 0; i < count; i++ {
		scanner.Scan()
		input, _ := strconv.Atoi(scanner.Text())
		if input == 0 {
			lastIndex := len(stack)-1
			total -= stack[lastIndex]
			stack = stack[:lastIndex]
		} else {
			stack = append(stack, input)
			total += input
		}
	}
	fmt.Print(total)
}

File: BOJ/10773_제로/kotlin/Main.kt
import java.util.Scanner
import java.util.Stack

fun main() {
    val scanner = Scanner(System.`in`)
    val k = scanner.nextInt()

    val stack = Stack<Int>()
    var input: Int

    repeat(k) {
        input = scanner.nextInt()
        if (input == 0) {
            stack.pop()
        } else {
            stack.push(input)
        }
    }
    println(stack.sum())
}
